src/dataIntegration/transcriptomics/EFlux/eFlux.m

```diff
@@ -0,0 +1,142 @@
+%                             * softBounds:       Whether to use soft bounds for the infered constraints or to add flexibility variables (default: false).
+%                             * weightFactor:     The weight factor for soft bounds (default: 1) 
+% OUTPUT:
+%    foldChange:             The fold change between the objective of the
+%                            condition and the objective of the control
+%                            expression
+%    standardError:          The error if noise is being used.
+%    solControl:             The solution of the given Control expression;
+%    solCondition:           The solution of the given Condition expression;
+%
+% ..Author:    Thomas Pfau OCt 2018
+%
+% NOTE:
+%    This si an implementation of the eFlux concept as presented in:
+%    Interpreting Expression Data with Metabolic Flux Models: Predicting Mycobacterium tuberculosis Mycolic Acid Production
+%    Colijn C, Brandes A, Zucker J, Lun DS, Weiner B, et al. (2009)
+%    PLOS Computational Biology 5(8): e1000489. https://doi.org/10.1371/journal.pcbi.1000489
+%    Please note, that this code does not perform any preprocessing expcept
+%    for that described in the above paper after array normalization. 
+
+normFun = @(mean,std) normrnd(mean,std);
+
+if ~isfield(controlExpression,'preprocessed')
+    controlExpression.preprocessed = true;
+end
+
+parser = inputParser();
+parser.addRequired('model',@(x) verifyModel(x,'simpleCheck',true));
+parser.addRequired('controlExpression',@(x) verifyEFluxExpressionStruct(model,x));
+parser.addRequired('conditionExpression',@(x) verifyEFluxExpressionStruct(model,x));
+parser.addParameter('testNoise',false, @(x) islogical(x) || (isnumeric(x) && (x==1 || x==0)));
+parser.addParameter('noiseCount',10, @isnumeric);
+parser.addParameter('noiseFun',normFun, @(x) isa(x, 'function_handle'));
+parser.addParameter('noiseStd',[], @isnumeric);
+parser.addParameter('controlData',[], @(x) verifyEFluxExpressionStruct(model,x));
+parser.addParameter('minSum',false,@(x) islogical(x) || (isnumeric(x) && (x==1 || x==0)));
+parser.addParameter('softBounds',false,@(x) islogical(x) || (isnumeric(x) && (x==1 || x==0)));
+parser.addParameter('weightFactor',1,@isnumeric);
+
+
+parser.parse(model,controlExpression,conditionExpression,varargin{:});
+testNoise = parser.Results.testNoise;
+noiseCount = parser.Results.noiseCount;
+noiseStd = parser.Results.noiseStd;
+noiseFun = parser.Results.noiseFun;
+controlData = parser.Results.controlData;
+noisyControl = [];
+% remove enforcing bounds (contradict eFlux concept)
+if(any(model.lb > 0 | model.ub < 0))
+    warning('Enforcing bounds for the following fluxes have been removed:\n%s', strjoin(model.rxns((model.lb > 0 | model.ub < 0)),'\n'));
+    model.lb(model.lb > 0) = 0;
+    model.ub(model.ub < 0) = 0;
+end
+    
+if testNoise
+    noisyControl = repmat(columnVector(controlExpression.value),1,noiseCount);
+    if isempty(controlData) && isempty(noiseStd) 
```

Comment:
The check of `isempty(controlData)` that throws an error shall be placed before `if testNoise`. Then, the repetitive `if ~isempty(controlData)` conditions  can be omitted.

Reply:
I don't really get this. 
missing/empty controlData will only throw an error if testNoise is active. 
I'm not sure how you want this changed.

Reply:
ok